packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2828,31 +2829,71 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
+    private async waitForDeltaManagerToCatchup(snapShot: ISnapshotTree,
+        summaryRefSeq: number,
+        readAndParseBlob: ReadAndParseBlob,
+        summaryLogger: ITelemetryLogger,
+        ): Promise<void> {
+        const latestSnapshotRefSeq = await seqFromTree(snapShot, readAndParseBlob);
+        if (latestSnapshotRefSeq !== summaryRefSeq) {
```

Comment:
I'm not sure how summaryRefSeq is relevant here, and thus this check.
It seems like we have to assert that latestSnapshotRefSeq  >= summaryRefSeq (as fetching latest summary after seeing an ack should give us summary at the same seq# or higher). If they are equal, then latestSnapshotRefSeq < this.deltaManager.lastSequenceNumber (because reference seq number of a summary is below it's ack seq number) and op waiting below would be noop.

So I'd rather see this code only waiting up to specific sequence number, and move the code RE getting summary ref sequence number and assert to the caller. And then I'd rather make waitForSeq() work properly when it is called in condition when we already reached target sequence number (today it does not, i.e. calling waitForSeq(0) will return only when _some_ new op shows up, which seems wrong).